marshal.go

```diff
@@ -172,6 +172,11 @@ func Marshal(info TypeInfo, value interface{}) ([]byte, error) {
 		return marshalDate(info, value)
 	case TypeDuration:
 		return marshalDuration(info, value)
+	case TypeCustom:
+		switch info.(type) {
+		case VectorType:
+			return marshalVector(info.(VectorType), value)
+		}
```

Comment:
you can simplify this block of code by assigning the type to var so you eliminate the need for the second type assertion:
```suggestion
	case TypeCustom:
		switch t := info.(type) {
		case VectorType:
			return marshalVector(t, value)
		}
	}
```

Reply:
Also the switch statement appears to only handle the `VectorType` case without any `default` case.

Reply:
The nested switch looks redundant.
Maybe something like this would be more suitable?
```
case TypeCustom:
	if vector, ok := info.(VectorType); ok {
		return marshalVector(vector, value)
	}
```

Reply:
Done.